### PR TITLE
fix(help command): RHMAP-19895 - Fix issues and allow use fhc command…

### PR DESCRIPTION
**Pull request by camilamacedo86**
…s --help

### CODE REVIEW

**Comment by damienomurchu** [comment]
Just verified this according to the verification steps above. All seems to function as expected. Just want to test around edge cases and will approve then if all ok :)

**Comment by camilamacedo86** [comment]
@damienomurchu tks for the collaboration 👍 🥇

**Comment by damienomurchu** [comment]
@camilamacedo86 there seems to be a regression introduced with these changes where help output is no longer been output from the below commands:

- fhc 4.25 (latest published version):
```
╭─ ~/work/fh-fhc ‹706f3eb*› 
╰─$ fhc ping --help                                                1 ↵
Options:
  --help     Show help                                        [boolean]
  --version  Show version number                              [boolean]

fhc ERR! cb() never called!
fhc ERR!  
fhc Command executed with error
╭─ ~/work/fh-fhc ‹706f3eb*› 
╰─$ fhc ping help  
Usage:
 fhc ping --app=<app> --env=<env>

Options:
  --help     Show help                                        [boolean]
  --version  Show version number                              [boolean]
  --app, -a  Unique 24 character GUID of your cloud application.
                                                             [required]
  --env, -e  Unique 24 character GUID of the environment where this
             cloud application is deployed.                  [required]

Examples:
  fhc ping --app=<app>                 Ping the cloud application
  --env=<environment>                  <app-id>

Missing required argument: env
fhc ERR! YError: Missing required argument: env
fhc ERR! 
fhc ERR! System Linux 4.14.11-200.fc26.x86_64
fhc ERR! command "/home/damienmurphy/.nvm/versions/node/v6.9.5/bin/node" "/home/damienmurphy/.nvm/versions/node/v6.9.5/bin/fhc" "ping" "help"
fhc Command executed with error
```

- fhc changes from this PR
```
╭─ ~/work/fh-fhc ‹706f3eb*› 
╰─$ node ./bin/fhc.js ping --help                                                                   
Ping a cloud app
╭─ ~/work/fh-fhc ‹706f3eb*› 
╰─$ node ./bin/fhc.js help ping  
Ping a cloud app
```

**Comment by camilamacedo86** [comment]
@damienomurchu it is not a regression it is a fix. 

When the command  `node ./bin/fhc.js ping --help` runs should show the help/desc of the ping command. In the published version it is returning the --help desc instead of the ping command desc which was fixed in this pr. 

c/c @davidffrench

**Comment by davidffrench** [comment]
@camilamacedo86 Unfortunately JIRA is currently down so I can't look at the original ticket opened by peter.

The expectation of running --help is to see what the usage and parameters should be. I would expect and a normal user would expect to see the following on a --help
```
Usage:
 fhc ping --app=<app> --env=<env>

Options:
  --help     Show help                                        [boolean]
  --version  Show version number                              [boolean]
  --app, -a  Unique 24 character GUID of your cloud application.
                                                             [required]
  --env, -e  Unique 24 character GUID of the environment where this
             cloud application is deployed.                  [required]
```

**Comment by camilamacedo86** [comment]
@davidffrench and @damienomurchu  thank you for the collab. I catcher what you mean. 
It is fixed now. Please, feel free to re-check.

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "tar": "2.2.1",
     "underscore": "1.8.3",
     "underscore-deep-extend": "0.0.5",
-    "yargs": "9.0.1"
+    "yargs": "12.0.1"
```
Comment (damienomurchu) [inline package.json]: Is there any breaking changes introduced in the major bumps here?

File: bin/fhc.js [hunk 2]
```diff
@@ -17,3 +17,22 @@
     });
   });
 })();
+
+/**
+ * This fuction was created in order to apply the command fhc help <commands> when the
+ * input is fhc commands --help or fhc commands help. It is chekcing if the last position is the --help
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply bin/fhc.js]: 👍 Done

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,7 +32,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Came back the command "fhc keys user add <label>" which was missing.
 - Add unit tests for the command "fhc keys user add <label>"
 
-
+#Released versions
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply CHANGELOG.md]: 👍 Done

File: package.json (hunk 1 above)
Comment (camilamacedo86) [reply package.json]: "Upgrade version of yargs from 9.0.1 to 12.0.1 ( it doesn't have break changes which affect this project - it will just get bug fix and allow new features of it). The usage of the yargs is to run the commands and its usage is re-checked in this PR." (ref first commit)

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+#Unreleased versions
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply CHANGELOG.md]: 👍 Done

**Comment by camilamacedo86** [comment]
@damienomurchu changes requested made. Regards the break changes just the support of the NodeJS 4 which is deprecated that affects this project since in the eng it was defined to versions >= 4.4 then it was changed and the bumped version now is 6.0.0.

**Comment by camilamacedo86** [comment]
Hi folks (@damienomurchu, @davidffrench, @austincunningham)

The changes requested here were made. Please, could you check this one? Could we move forward?
